Scripts/Player.cs

```diff
@@ -135,44 +105,53 @@ public override void _PhysicsProcess(float delta)
 
     public void Move(Vector2 weightedDirection, float delta)
     {
-        float targetSpeed = IsSprinting ? MaxSpeed : WalkingSpeed;
-        Vector3 targetVelocity = targetSpeed * new Vector3(weightedDirection.x, 0, weightedDirection.y);
-        Vector3 targetAcceleration = targetVelocity - velocity;
-        Vector3 actualAcceleration = targetAcceleration;
+        // Add acceleration based on user input
+        float accFactor = IsSprinting ? SprintAccelerationFactor : WalkAccelerationFactor;
+        Vector3 direction = new Vector3(weightedDirection.x, 0, weightedDirection.y);
+        AddCentralForce(accFactor * Mass * direction);
 
-        // Player trying to accelerate in the same direction
-        if (targetVelocity.Dot(targetAcceleration) > 0)
-        {
-            if (targetAcceleration.Length() > MaxAcceleration)
-            {
-                actualAcceleration = targetAcceleration.Normalized() * MaxAcceleration;
-            }
-        }
-        // Player trying to slow down/switch direction
-        else
-        {
-            if (targetAcceleration.Length() > MaxDeceleration)
-            {
-                actualAcceleration = targetAcceleration.Normalized() * MaxDeceleration;
-            }
-        }
+        // Add drag to cap the maximum speed, proportional to square of speed
+        Vector3 velocity = LinearVelocity * new Vector3(1, 0, 1);
+        Vector3 velDirection = velocity.Length() > 1 ? velocity.Normalized() : velocity;

```

Comment:
Intended for the length to be double-counted when length <= 1?

Reply:
Yeah, speed squared.